pkg/controller/upgradeconfig/cluster_upgrader.go

```diff
@@ -588,6 +590,9 @@ func performClusterHealthCheck(c client.Client) (bool, error) {
 	promurl := "https://" + route.Spec.Host + "/api/v1/query"
 
 	req, err := http.NewRequest("GET", promurl, nil)
+	if err != nil {
+		return false, err
```

Comment:
For each of these where it's returning false and err, would prefer the error include context of what caused it rather than just returning the error. 

Eg.
```suggestion
		return false, fmt.Errorf("Could not query Prometheus (reason: %s)", err)
```

Reply:
Thanks a lot for the review Matt ! Have submitted changes..